examples/Example7/example7.cu

```diff
@@ -0,0 +1,233 @@
+{
+  // --- Create materials.
+  G4Material *galactic = G4NistManager::Instance()->FindOrBuildMaterial("G4_Galactic");
+  G4Material *silicon  = G4NistManager::Instance()->FindOrBuildMaterial("G4_Si");
+  //
+  // --- Define a world.
+  G4double worldDim         = 1 * m;
+  G4Box *worldBox           = new G4Box("world", worldDim, worldDim, worldDim);
+  G4LogicalVolume *worldLog = new G4LogicalVolume(worldBox, galactic, "world");
+  G4PVPlacement *world      = new G4PVPlacement(nullptr, {}, worldLog, "world", nullptr, false, 0);
+  // --- Define a box.
+  G4double boxDim             = 0.5 * m;
+  G4double boxPos             = 0.5 * boxDim;
+  G4Box *siliconBox           = new G4Box("silicon", boxDim, boxDim, boxDim);
+  G4LogicalVolume *siliconLog = new G4LogicalVolume(siliconBox, silicon, "silicon");
+  new G4PVPlacement(nullptr, {boxPos, boxPos, boxPos}, siliconLog, "silicon", worldLog, false, 0);
+  //
+  // --- Create particles that have secondary production threshold.
+  G4Gamma::Gamma();
+  G4Electron::Electron();
+  G4Positron::Positron();
+  G4Proton::Proton();
+  G4ParticleTable *partTable = G4ParticleTable::GetParticleTable();
+  partTable->SetReadiness();
+  //
+  // --- Create production - cuts object and set the secondary production threshold.
+  G4ProductionCuts *productionCuts = new G4ProductionCuts();
+  constexpr G4double ProductionCut = 1 * mm;
+  productionCuts->SetProductionCut(ProductionCut);
+  //
+  // --- Register a region for the world.
+  G4Region *reg = new G4Region("default");
+  reg->AddRootLogicalVolume(worldLog);
+  reg->UsedInMassGeometry(true);
+  reg->SetProductionCuts(productionCuts);
+  //
+  // --- Update the couple tables.
+  G4ProductionCutsTable *theCoupleTable = G4ProductionCutsTable::GetProductionCutsTable();
+  theCoupleTable->UpdateCoupleTable(world);
+}
+
+__constant__ __device__ struct G4HepEmParameters g4HepEmPars;
+__constant__ __device__ struct G4HepEmData g4HepEmData;
+
+static G4HepEmRunManager *InitG4HepEm()
+{
+  G4HepEmRunManager *runManager = new G4HepEmRunManager(true);
+  // We don't need the G4HepEmRunManager to have a RNG.
+  runManager->Initialize(nullptr, 0);
+
+  G4HepEmData *data          = runManager->GetHepEmData();
+  G4HepEmMatCutData *cutData = data->fTheMatCutData;
+  G4cout << "fNumG4MatCuts = " << cutData->fNumG4MatCuts << ", fNumMatCutData = " << cutData->fNumMatCutData << G4endl;
+
+  // Copy to GPU.
+  COPCORE_CUDA_CHECK(cudaMemcpyToSymbol(g4HepEmPars, runManager->GetHepEmParameters(), sizeof(G4HepEmParameters)));
+  CopyG4HepEmDataToGPU(data);
+
+  // Create G4HepEmData with the device pointers.
+  G4HepEmData dataOnDevice;
+  dataOnDevice.fTheMatCutData   = data->fTheMatCutData_gpu;
+  dataOnDevice.fTheMaterialData = data->fTheMaterialData_gpu;
+  dataOnDevice.fTheElementData  = data->fTheElementData_gpu;
+  dataOnDevice.fTheElectronData = data->fTheElectronData_gpu;
+  dataOnDevice.fThePositronData = data->fThePositronData_gpu;
+  // fTheSBTableData missing for now.
```

Comment:
It's available now on the device in `G4HepEm`. However,  this data is used only in the SB bremsstrahlung  interaction model, as we discussed, so I just wanted to let you know.

Reply:
Great, didn't see yesterday's commit! Yes, I'll probably start with final state sampling of ionization and keep you posted on how that works out...